server/backend/check.ml

```diff
@@ -232,19 +243,35 @@ let get_obuilder ~conf ~opam_repo ~opam_repo_commit ~extra_repos switch =
   in
   let open Obuilder_spec in
   let cache = cache ~conf in
-  let from = match Server_configfile.platform_os conf with
-    | "linux" -> Server_configfile.platform_image conf
+  let os = Server_configfile.platform_os conf in
+  let distribution = Server_configfile.platform_distribution conf in
+  let from = match os with
+    | "linux" -> "ocaml/opam:"^distribution (* typically this is 'debian-unstable' which is 5.0.0 *)
+    | "freebsd" -> distribution^"-ocaml-5.0" (* Either 4.14.1 or 5.0.0 could be selected *)
+    | "macos" -> "macos-"^distribution^"-ocaml-5.0" (* Either 4.14.1 or 5.0.0 could be selected *)
```

Comment:
Actually is there a way to avoid being OCaml version specific here and have a shortcut from e.g. `macos-homebrew` to `macos-homebrew-ocaml-5.0`

This way we don't have to change opam-health-check everytime there is a new version of the compiler